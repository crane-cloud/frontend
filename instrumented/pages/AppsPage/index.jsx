import React from "react";
import { withRouter } from "react-router-dom";
import { connect } from "react-redux";
import PropTypes from "prop-types";
// import { v4 as uuidv4 } from "uuid";

import AppsList from "../../components/AppsList";
import CreateApp from "../../components/CreateApp";
import "./AppsPage.css";
import DashboardLayout from "../../components/Layouts/DashboardLayout";

class AppsPage extends React.Component {
  constructor(props) {
    super(props);
    this.initialState = {
      name: "",
      openModal: false, // add project component is closed initially
      error: "",
      port: "",
      word: "",
    };

    this.state = this.initialState;

    this.showForm = this.showForm.bind(this);
    this.hideForm = this.hideForm.bind(this);
    this.getProjectDetails = this.getProjectDetails.bind(this);
    this.handleCallbackSearchword = this.handleCallbackSearchword.bind(this);
  }

  // componentDidMount() {
    
  // }

  // componentDidUpdate(prevProps) {
  
  // }

  handleCallbackSearchword(word) {
    this.setState({ word: word });
  }

  getProjectDetails(projects, id) {
    return projects?.find((project) => project.id === id);
  }

  showForm() {
    this.setState({ openModal: true });
  }

  hideForm() {
    this.setState(this.initialState);
  }

  render() {
    const { openModal, word } = this.state;

    const {
      match: { params },
      projects,
    } = this.props;

    const { projectID } = params;
    const projectDetails = this.getProjectDetails(projects, projectID);

    const filteredDetails = {
      name: projectDetails?.name,
      description: projectDetails?.description,
      organisation: projectDetails?.organisation,
      project_type: projectDetails?.project_type,
    };

    localStorage.setItem("project", JSON.stringify(filteredDetails));

    return (
      <div>
        {openModal ? (
          <DashboardLayout
            name={projectDetails?.name}
            header="Create App"
            showBtn
            buttontext="Close"
            btntype="close"
            btnAction={this.hideForm}
          >
            <CreateApp params={params} />
          </DashboardLayout>
        ) : (
          <DashboardLayout
            name={projectDetails?.name}
            header="Apps"
            showBtn
            buttontext="+ New App"
            showSearchBar
            placeholder="Search through apps"
            searchAction={this.handleCallbackSearchword}
            btnAction={this.showForm}
          >
            <AppsList
              params={params}
              word={word}
              openComponent={this.showForm}
            />
          </DashboardLayout>
        )}
      </div>
    );
  }
}

AppsPage.propTypes = {
  projects: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
};

export const mapStateToProps = (state) => {
  const { projects } = state.userProjectsReducer;
  return {
    projects,
  };
};

const mapDispatchToProps = {
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(withRouter(AppsPage));
