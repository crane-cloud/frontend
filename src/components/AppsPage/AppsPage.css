.BackDiv img {
  display: inline;
}

.BackDiv {
  display: inline;
  color: black;
}

.BackDiv p {
  display: inline;
}
.BackDiv img:hover {
  transition: 0.2s;
  transform: scale(1.2);
}

/*! ////////////////////////////////////////// */
.ModalForm.AddAppModal .ModalFormHeading {
  margin-bottom: 0;
}

.ModalFormInputsBasic {
  display: grid;
  gap: 1rem;
}

.ModalFormInputsBasic input {
  max-width: 25rem;
}

.ModalFormInputsBasic input::placeholder {
  color: gray;
}

.EnvVarsInputGroup {
  width: 100%;
  max-width: 25rem;
  display: grid;
  grid-template-columns: 3fr 1fr;
  gap: 0.5rem;
}

.EnvVarsInputGroup .EnvVarsInputs {
  width: 70%;
  max-width: 25rem;
  display: grid;
  gap: 0.5rem;
  grid-template-columns: 1fr 1fr;
}

.EnvVarsInputGroup .EnvVarsAddBtn {
  position: relative;
  display: flex;
  align-items: center;
}

.EnvVarsInputGroup .EnvVarsAddBtn button.EnvVarAddBtn {
  position: absolute;
  max-width: 100%;
  color: #fff;
  background-color: #008ac1;
  outline: 2px solid #008ac1;
}

.EnvVarsInputGroup .EnvVarsAddBtn button.EnvVarAddBtn:hover {
  color: #008ac1;
  background-color: #fff;
}

.EnvVarsInputGroup input {
  max-width: 10rem;
}

.EnvVarsInputGroup input::placeholder {
  color: grey;
}

.ModalFormButtons {
  width: 100%;
}

.EnvVarsTable {
  width: 100%;
  margin-bottom: 1rem;
}

.EnvVarsTable table {
  max-width: 25rem;
  text-align: left;
}

.EnvVarsTable table thead td {
  font-weight: normal;
  color: #555;
}

.EnvVarsTable table thead td:last-child {
  width: 4rem;
  text-align: right;
}

.EnvVarsTable table td:last-child {
  width: 4rem;
  display: block;
  margin: auto;
}

.EnvVarsTable table td {
  font-weight: normal;
}

.EnvVarsTable table td img {
  margin: auto;
}

.EnvVarsTable table td img:hover {
  /* cursor: pointer; */
  filter: invert(42%) sepia(93%) saturate(1352%) hue-rotate(180deg)
    brightness(119%) contrast(5%);
}

.EnvVarsTable thead,
.EnvVarsTable tbody tr {
  display: table;
  width: 100%;
  table-layout: fixed;
}

/*- //////////////////////////////////////////////////// */

.ModalForm.AddAppModal {
  display: grid;
  gap: 2rem;
  height: 100%;
}

.ModalForm.AddAppModal input {
  height: 3.1rem;
}

.ModalForm.AddAppModal .ModalFormHeading {
  flex: 0 1 auto;
}

.ModalForm.AddAppModal .ModalFormInputs {
  padding-right: 10px;
  flex: 1 0 auto;
  overflow-y: auto;
}

.ModalForm.AddAppModal .ModalFormButtons.AddAddButtons {
  flex: 0 1 40px;
}

.ModalForm.AddAppModal .ModalFormInputs::-webkit-scrollbar {
  width: 4px;
  height: 0px;
}
.ModalForm.AddAppModal .ModalFormInputs::-webkit-scrollbar-track {
  background: #f1f1f1;
}
.ModalForm.AddAppModal .ModalFormInputs::-webkit-scrollbar-thumb {
  background: #555;
}
.ModalForm.AddAppModal .ModalFormInputs::-webkit-scrollbar-thumb:hover {
  background: #000;
}

.HeadingWithTooltip {
  display: flex;
}

.DbSupportSection {
  display: grid;
  grid-template-columns: 1fr;
  gap: 0%;
  padding-bottom: 1rem;
}

.HeadingWithTooltip h4 {
  margin: 0;
  padding: 0.5rem 0;
}

.DbSupportCheckField,
.PrivateImageCheckField {
  display: flex;
}

.InputFieldWithTooltip {
  max-width: 25rem; /* same as input field*/
  display: flex;
  position: relative;
}

.InputTooltipContainer {
  position: absolute;
  right: 0;
  padding-right: 0.5rem;
  height: 100%;
  display: flex;
  align-items: center;
}

.DatabaseSupportTabContainer,
.PrivateImageTabContainer {
  max-width: 25rem; /* this is to correspond with the width of the input elements */
}

.PrivateImageInputs,
.DatabaseSupportInputs {
  display: grid;
  gap: 0.5rem;
}

.PrivateImageInputs input,
.DatabaseSupportInputs input {
  width: auto;
}

.DbCheckBoxInputGroup {
  display: grid;
  gap: 1rem;
}

.ConnectModal {
  width: 82.19%;
  margin-left: 17.81%;
}


.AppsPage {
  padding: 2rem;
}