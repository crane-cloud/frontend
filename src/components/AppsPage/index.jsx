import React from "react";
import { withRouter } from "react-router-dom";
import { connect } from "react-redux";
import PropTypes from "prop-types";
// import { v4 as uuidv4 } from "uuid";
import createApp, { clearState } from "../../redux/actions/createApp";
import InformationBar from "../InformationBar";
import AppsList from "../AppsList";
import Header from "../Header";
import SideBar from "../SideBar";
import CreateApp from "../CreateApp";
import "./AppsPage.css";

class AppsPage extends React.Component {
  constructor(props) {
    super(props);
    this.initialState = {
      name: "",
      openModal: false, // add project component is closed initially
      error: "",
      port: "",
      word: "",
    };

    this.state = this.initialState;

    this.showForm = this.showForm.bind(this);
    this.hideForm = this.hideForm.bind(this);
    this.getProjectDetails = this.getProjectDetails.bind(this);
    this.handleCallbackSearchword = this.handleCallbackSearchword.bind(this);
  }

  componentDidMount() {
    const { clearState } = this.props;
    clearState();
  }

  componentDidUpdate(prevProps) {
    const { isCreated } = this.props;

    if (isCreated !== prevProps.isCreated) {
      this.hideForm();
    }
  }
  handleCallbackSearchword(word) {
    this.setState({ word: word });
  }

  getProjectDetails(projects, id) {
    const project = projects.find((project) => project.id === id);
    return project;
  }

  showForm() {
    this.setState({ openModal: true });
  }

  hideForm() {
    const { clearState } = this.props;
    clearState();
    this.setState(this.initialState);
  }

  render() {
    const { openModal, word } = this.state;

    const {
      match: { params },
      isCreated,
      projects,
    } = this.props;

    const { projectID } = params;
    const projectDetails = this.getProjectDetails(projects, projectID);

    const filteredDetails = {
      name: projectDetails.name,
      description: projectDetails.description,
      organisation: projectDetails.organisation,
      project_type: projectDetails.project_type,
    };

    localStorage.setItem("project", JSON.stringify(filteredDetails));

    return (
      <div className="Page">
        <div className="TopBarSection">
          <Header />
        </div>
        <div className="MainSection">
          <div className="SideBarSection">
            <SideBar
              name={projectDetails.name}
              params={params}
              description={projectDetails.description}
              pageRoute={this.props.location.pathname}
              allMetricsLink={`/projects/${projectID}/metrics`}
              cpuLink={`/projects/${projectID}/cpu/`}
              memoryLink={`/projects/${projectID}/memory/`}
              databaseLink={`/projects/${projectID}/databases`}
              networkLink={`/projects/${projectID}/network/`}
            />
          </div>
          {openModal ? (
            <CreateApp closeComponent={this.hideForm} params={params} />
          ) : (
            <div className="MainContentSection">
              <div className="InformationBarSection">
                <InformationBar
                  header="Apps"
                  showBtn
                  showSearchBar
                  placeholder="Search through apps"
                  searchAction={this.handleCallbackSearchword}
                  btnAction={this.showForm}
                />
              </div>
              <div className="ContentSection AppsPage SmallContainer">
                <AppsList
                  params={params}
                  newAppCreated={isCreated}
                  word={word}
                />
              </div>
            </div>
          )}
        </div>
      </div>
    );
  }
}

AppsPage.propTypes = {
  isCreated: PropTypes.bool.isRequired,
  clearState: PropTypes.func.isRequired,
  projects: PropTypes.arrayOf(PropTypes.shape({})).isRequired,
};

const mapStateToProps = (state) => {
  const { isCreated } = state.createAppReducer;
  const { projects } = state.userProjectsReducer;
  return {
    isCreated,
    projects,
  };
};

const mapDispatchToProps = {
  createApp,
  clearState,
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(withRouter(AppsPage));
