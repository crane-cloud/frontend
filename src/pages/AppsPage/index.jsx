import React, { useState } from "react";
import { useSelector } from "react-redux";
import { useParams, useLocation } from "react-router-dom";
import PropTypes from "prop-types";

import AppsList from "../../components/AppsList";
import CreateApp from "../../components/CreateApp";
import "./AppsPage.css";
import DashboardLayout from "../../components/Layouts/DashboardLayout";

const AppsPage = () => {
  
  const [word, setWord] = useState("");

  const location = useLocation(); 

  const queryParams = new URLSearchParams(location.search); 
  const initialOpenModal = queryParams.get("initialOpenModal") === "true"; 
  const [openModal, setOpenModal] = useState(initialOpenModal);

  const { projectID } = useParams();

  const projects = useSelector((state) => state.userProjectsReducer.projects);

  const getProjectDetails = (projects, id) => {
    return projects?.find((project) => project.id === id);
  };

  const projectDetails = getProjectDetails(projects, projectID);

  const filteredDetails = {
    name: projectDetails?.name,
    description: projectDetails?.description,
    organisation: projectDetails?.organisation,
    project_type: projectDetails?.project_type,
  };

  localStorage.setItem("project", JSON.stringify(filteredDetails));

  const showForm = () => setOpenModal(true);
  const hideForm = () => {
    setOpenModal(false);
  };

  const handleCallbackSearchword = (word) => {
    setWord(word);
  };

  return (
    <div>
      {openModal ? (
        <DashboardLayout
          name={projectDetails?.name}
          header="Create App"
          showBtn
          buttontext="Close"
          btntype="close"
          btnAction={hideForm}
        >
          <CreateApp params={{ projectID }} />
        </DashboardLayout>
      ) : (
        <DashboardLayout
          name={projectDetails?.name}
          header="Apps"
          showBtn
          buttontext="+ New App"
          showSearchBar
          placeholder="Search through apps"
          searchAction={handleCallbackSearchword}
          btnAction={showForm}
        >
          <AppsList
            params={{ projectID }}
            word={word}
            openComponent={showForm}
          />
        </DashboardLayout>
      )}
    </div>
  );
};

AppsPage.propTypes = {
  initialOpenModal: PropTypes.bool, 
};

export default AppsPage;
