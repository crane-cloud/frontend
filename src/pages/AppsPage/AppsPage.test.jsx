import React from "react";
import { shallow } from "enzyme";
import { MemoryRouter } from "react-router-dom"; 
import AppsPage from "./";

describe("testing the Apps page component", () => {
  it("this matches Apps page component snapshot", () => {
    const mycomponent = shallow(
      <MemoryRouter initialEntries={["/your-path"]}> 
        <AppsPage />
      </MemoryRouter>
    );
    expect(mycomponent).toMatchSnapshot();
  });

  it("should match wrapped component snapshot", () => {
    const wrapper = shallow(
      <MemoryRouter initialEntries={["/your-path"]}> 
        <AppsPage />
      </MemoryRouter>
    );
    expect(wrapper).toBeDefined();
  });
});
