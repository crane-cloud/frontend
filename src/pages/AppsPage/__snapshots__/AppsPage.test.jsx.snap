// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the Apps page component this matches Apps page component snapshot 1`] = `ShallowWrapper {}`;
